### PR TITLE
Add documentation for binary content extraction and ingest pipelines

### PR DESCRIPTION
**Pull request by navarone-feekery**
### Summary

- Add BINARY_CONTENT_EXTRACTION.md
- Add INGEST_PIPELINES.md
- Add `binary_` prefix to binary content extraction config fields to reduce confusion (it is unrelated to content extraction from HTML/XML)

#### Pre-Review Checklist
- [x] This PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `crawler.yml.example` and `elasticsearch.yml.example`)
- [x] This PR has a meaningful title
- [x] This PR links to all relevant GitHub issues that it fixes or partially addresses
    - If there is no GitHub issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v0.1.0`)
- [x] Considered corresponding documentation changes
- [x] Contributed any configuration settings changes to the configuration reference
- [x] Ran `make notice` if any dependencies have been added

### Release note

The configuration field `content_extraction_enabled` was renamed to `binary_content_extraction_enabled`.
The configuration field `content_extraction_mime_types` was renamed to `binary_content_extraction_mime_types`.

### CODE REVIEW

**Review by seanstory — DISMISSED** [review]
LGTM. One nit. Also, should probably add a "release notes" section to cover the config name changes?

File: docs/features/BINARY_CONTENT_EXTRACTION.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Binary content is extracted by converting file contents to base64 and including the output in a document to index.
+This value is picked up by an [Elasticsearch ingest pipeline](https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest.html) that will convert the base64 content into plain text, to store in the `body` field of the same document.
+
+## Using this feature
+
+1. Enable ingest pipelines in the Elasticsearch configuration
+2. Enable binary content extraction in the Crawler configuration
+3. Select which MIME types should have their contents extracted
+   - The MIME type is determined by the HTTP response’s `Content-Type` header when downloading a given file
+   - While intended primarily for PDF and Microsoft Office formats, you can use any of the supported formats documented by [Apache Tika](https://tika.apache.org/)
```
Comment (marciw) [inline docs/features/BINARY_CONTENT_EXTRACTION.md:13]:
```suggestion
   - While intended primarily for PDF and Microsoft Office formats, you can use any of the formats supported by [Apache Tika](https://tika.apache.org/)
```

**Review by marciw — APPROVED** [review]
LGTM! one small suggestion

**Comment by navarone-feekery** [comment]
@seanstory @marciw thanks for the reviews! Unfortunately applying the change suggestions dismissed them. Can you re-review?

**Review by jedrazb — APPROVED** [review]

**Pull request merged**